Update `swift_import` docs and add warning (#1186)
bazelbuild/rules_swift: doc/rules.md

swift_import(<a href="#swift_import-name">name</a>, <a href="#swift_import-deps">deps</a>, <a href="#swift_import-data">data</a>, <a href="#swift_import-archives">archives</a>, <a href="#swift_import-module_name">module_name</a>, <a href="#swift_import-swiftdoc">swiftdoc</a>, <a href="#swift_import-swiftinterface">swiftinterface</a>, <a href="#swift_import-swiftmodule">swiftmodule</a>)
</pre>

Allows for the use of Swift textual module interfaces or precompiled Swift modules as dependencies in other
`swift_library` and `swift_binary` targets.

To use `swift_import` targets across Xcode versions and/or OS versions, it is required to use `.swiftinterface` files.

> These can be produced on the pre-built target if using Bazel with:

Reviewer: ```suggestion
These can be produced by the pre-built target if built with:
```

>   - `--features='swift.enable_library_evolution'`

Reviewer 2: The quotes aren't needed around the name of the feature.
